API/contacts/ContactAPI.php

```diff
@@ -0,0 +1,150 @@
+        }
+
+        /**
+         * Gets contact record which satisfies query.
+         * 
+         * @param string $query search query used for searching for set of records in the database.
+         * @param int $numOfResults max number of results that satisfy search query to be returned if search is successful.
+         * @param int $userID unique user identifier.
+         * @return array|false array of objects of the Contact class containing all information about each individual record or false if operation was unsuccessful.
+         */
+        public function GetContact(string $query, int $userID, int $numOfResults) : array|false
+        {
+            if ($this->mysql->connect_error != null)
+                return false;
+
+            $queryArray = explode(" ", $query);
+
+            $searchQuery = "";
+
+            foreach ($queryArray as $word) {
+                $searchQuery = $searchQuery . 
+                "firstName LIKE '%$word%' OR 
+                lastName LIKE '%$word%' OR
+                phone LIKE '%$word%' OR
+                email LIKE '%$word%'" . " OR ";
+            }
+
+            // Removes the last " OR " inside of the searchQuery
+            $searchQuery = substr($searchQuery, 0, strlen($searchQuery) - 4);
+        
+            $result = $this->mysql->query("SELECT * FROM Contacts WHERE ($searchQuery) AND userID=$userID LIMIT $numOfResults");
+        
+            if ($result == false)
+                return false;
+        
+            $resultArray = [];
+
+            while($record = $result->fetch_object())
+                $resultArray[] = Contact::Deserialize($record);
+                
+            return $resultArray;
+        }
+
+        /**
+         * Updates contact record.
+         * 
+         * @param object $contact contact object of the Contact class.
+         * @return object|false contact object of the Contact class containing updated information or false if operation was unsuccessful.
+         */
+        public function UpdateContact(object $contact) : object|false
+        {
+            if ($this->mysql->connect_error != null)
+                return false;
+        
+            $result = $this->mysql->query("UPDATE Contacts SET firstName='$contact->firstName', lastName='$contact->lastName', email='$contact->email', phone='$contact->phone' WHERE ID=$contact->ID");
+            
+            if ($result)
+                return $this->GetContactByID($contact->ID);
+
+            return false;
+        }
+
+        /**
+         * Deletes contact record.
+         * 
+         * @param object $contact contact object of the Contact class.
+         * @return bool true if operation was successful or false otherwise.
+         */
+        public function DeleteContact(object $contact) : bool
+        {
+            if ($this->mysql->connect_error != null)
+                return false;
+            
+            if ($this->GetContactByID($contact->ID) == false)
```

Comment:
For truthy/falsey values, should we use `===` and `!==`? Or is it ok to return if `GetContactByID` returns NULL for example?

Reply:
isn't === for javascript?

Reply:
nvm php also has it